python/ray/_private/runtime_env/packaging.py

```diff
@@ -592,7 +606,13 @@ async def download_and_unpack_package(
                     pkg_uri.encode(), namespace=None, timeout=None
                 )
                 if code is None:
-                    raise IOError(f"Failed to fetch URI {pkg_uri} from GCS.")
+                    raise IOError(
+                        f"Failed to download runtime_env file package {pkg_uri} "
```

Comment:
I think the way would be to examine the logs, `gcs_server.out` has the logs of the form `[runtime env manager] ID to URIs table:
[runtime env manager] URIs reference table:`. I'm not sure how useful it would be for the user to check these logs; this is more likely to be an internal error than something the user can fix.  After https://github.com/ray-project/ray/pull/24719, we haven't heard any reports of this error (but there hasn't been a stable Ray release since then, so we can't be sure that it's been fixed yet)